### PR TITLE
add event count for analysis

### PR DESCRIPTION
**Pull request by wangben718**
This PR is for events count.
image

### CODE REVIEW

**Comment by wangben718** [comment]
@burlave could you help me pass all the test cases in this branch?

File: R/avg.R [hunk 1]
```diff
@@ -55,6 +55,10 @@ avg_event <- function(id, group, par = NULL) {
       function(x) sd(x$Freq, na.rm = TRUE) / sqrt(nrow(x)),
       FUN.VALUE = numeric(1)
     )
+    count <- vapply(split(res, res$Var2),
+      function(x) nrow(x),
```
Comment (nanxstats) [inline R/avg.R:59]: Is `vapply(..., function(x) nrow(x))` equivalent to `vapply(..., nrow)`?

Comment (wangben718) [reply R/avg.R:59]: > Is `vapply(..., function(x) nrow(x))` equivalent to `vapply(..., nrow)`?

It seems not correct. :)

Comment (nanxstats) [reply R/avg.R:59]: Care to show me an example when it's not equivalent? Would love to see it.

Comment (wangben718) [reply R/avg.R]: I made a mistake. It makes!

Comment (nanxstats) [reply R/avg.R]: OK good!

File: R/avg.R [hunk 2]
```diff
@@ -55,6 +55,10 @@ avg_event <- function(id, group, par = NULL) {
       function(x) sd(x$Freq, na.rm = TRUE) / sqrt(nrow(x)),
       FUN.VALUE = numeric(1)
     )
+    count <- vapply(split(res, res$Var2),
+      nrow,
```
Comment (fukuhiro2023) [inline R/avg.R:59]: This should be the of event counts per treatment group, right? If so, the following is suggested:
```suggestion
      function(x) sum(x$Freq, na.rm = TRUE), 
```
Currently, this seems to be total subject counts.

File: R/avg.R [hunk 3]
```diff
@@ -74,7 +78,8 @@ avg_event <- function(id, group, par = NULL) {
           group = unique(X$group),
           par = unique(X$par),
           avg = mean(X$n, na.rm = TRUE),
-          se = sd(X$n, na.rm = TRUE) / sqrt(nrow(X))
+          se = sd(X$n, na.rm = TRUE) / sqrt(nrow(X)),
+          count = nrow(X)
```
Comment (fukuhiro2023) [inline R/avg.R:82]: Same comment on Line 59. If this should be counts of events, the following is suggested:
```suggestion
          count = sum(X$n)
```

**Review by fukuhiro2023 — CHANGES_REQUESTED** [review]
I have checked based on an output, and there are 3 findings as below: 

1. The total number of events for each group seem not to be correct (highlighted by blue). As commented on the code, please check the logic of calculation.
2.  It seems that the number of events for each PT are not correctly displayed. For example, a count is more than 0 even though no subject has a PT (highlighted by yellow). This may be becuase an order of an output of `avg_event()` is different from that of `outdata$name`.
3. How about pudding blanks with 0 for event columns (highlighted by green)?

screen_shot_metalite ae_event

**Comment by fukuhiro2023** [comment]
Thank you for your update. My first finding has been resolved, but the second and third ones have not. Will these be handled as separate issues?

**Comment by fukuhiro2023** [comment]
To resolve my second finding, order of the event counts is aligned with `outdata$name`. Also, remove zero from unexpected rows, such as rows of population and no AE. Since the updated output looks good as below, there is no additional comment.
screen_shot_metalite ae_event_update

**Review by fukuhiro2023 — APPROVED** [review]

**Comment by wangben718** [comment]
Thanks @fukuhiro2023 for your findings.

**Pull request merged**